### ISSUE TITLE
请问如何在响应拦截器里修改响应内容

### ISSUE THREAD

**Issue by ZhengXiaowei** [flutterchina/dio#782]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |      |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

我想根据情况来调整返回的响应数据，拦截器里是这么写的：
```
@override
  Future onResponse(Response response) {
    Response response_copy = response;
    BaseResponse res = BaseResponse.fromJson(response_copy.data);
    if (res.code == 0)
      response.data = res.data;
    print("接口返回: ${response.data}");
    return super.onResponse(response);
  }
// BaseResponse.dart
class BaseResponse {
  int code;
  dynamic data;
  String msg;

  BaseResponse();
  // 省略
}
```
事实上在响应拦截器里打印的结果是想要的，但是使用的时候就会报`type '_InternalLinkedHashMap<String, dynamic>' is not a subtype of type 'Author'`的错误:

```
static Future<Response<Author>> getAuthor() async {
    return await dio.get("/api/author");
}

// Author.dart;
class Author {
  int id;
  String name;
  // 省略
  Author();
}

// 使用
// 一直出错所以改成了dynamic
Future<dynamic> getName() async {
  dynamic author = await getAuthor();
  print("author name is ${author.name}");
}
```

所以想问问，具体是哪里出错了，感谢！
